components/scream/src/physics/shoc/shoc_main_impl.hpp

```diff
@@ -15,6 +15,49 @@ namespace shoc {
  * #include this file, but include shoc_functions.hpp instead.
  */
 
+template<typename S, typename D>
+Int Functions<S,D>::shoc_init(
+  const Int&                  nbot_shoc,
+  const Int&                  ntop_shoc,
+  const view_1d<const Spack>& pref_mid)
+{
+  // This function calculates the maximum number of levels
+  // in pbl from surface
+
+  using ExeSpace = typename KT::ExeSpace;
+  view_1d<Int> npbl_d("npbl",1);
+
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, 1);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+
+    const Scalar pblmaxp = SC::pblmaxp;
+
+    int npbl_val = 0;
+
+    const int begin_pack_indx = ntop_shoc/Spack::n;
+    const int end_pack_indx   = nbot_shoc/Spack::n+1;
+    Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin_pack_indx, end_pack_indx),
```

Comment:
A reduction that sums over the 1-mask of levels satisfying pref_mid >= pblbmaxp is fine, but perhaps a more natural alternative would be a reduction on the max level. Just mentioning it in case you don't know you can do team-level max-reductions.

Reply:
From the fortran code 
```
   npbl = 0
   do k=nbot_shoc,ntop_shoc,-1
      if (pref_mid(k) >= pblmaxp) then
         npbl = npbl + 1
      end if
   end do
   npbl = max(npbl,1)
```
A max-reduction assume that `pref-mid` is strictly increasing? Is this the case? If so, then I agree a max-reduction is more natural and I will change the bfb test to have an increasing `pref_mid` (currently it is only random values).

Reply:
Yes, pref_mid is monotonically increasing. Also, note that the use of npbl later, e.g., in loops of the sort `k=nlev-1,nlev-npbl+1,-1`, implies npbl refers to a contiguous set of levels extending from the surface.

Reply:
@ambrad changes done. Now calculating "max levels from surface in pbl" and changed the test to have monotonic pref_mid.

Reply:
Are you testing the case that the reduction would never update npbl_val? I believe the value you initialize of `npbl_val = 1` at line 35 will not be seen in the parallel_reduce, so on exit npbl_val would be undefined in the case that the F90 would give 0.